ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_CppSamples/ZipHelper.cpp

```diff
@@ -0,0 +1,715 @@
+
+  Returns \c true if successful; otherwise \c false.
+*/
+
+bool ZipHelper::extractAll(const QString &outputPath)
+{
+    if (!zrOpen())
+    {
+#ifdef ZIP_LOGGING_ENABLED
+        qDebug() << "ZIP not open" << path();
+#endif
+        return false;
+    }
+
+    QDir outputDir(outputPath);
+
+#ifdef ZIP_LOGGING_ENABLED
+    qDebug() << outputPath << outputDir;
+#endif
+
+    return extractAll(outputDir);
+}
+
+// Methods moved from FileUtility to enable ZipHelper::setPath to work
+//--------------------------------------------------------------------
+
+#define kResourceScheme                         QStringLiteral("qrc")
+#define kResourcePathPrefix                     QStringLiteral(":/")
+
+#define kHttpScheme                             QStringLiteral("http")
+#define kHttpsScheme                            QStringLiteral("https")
+
+#define kPathHomePrefix                         QStringLiteral("~")
+#define kPathVarPrefix                          QStringLiteral("$")
+#define kPathVarHome                            QStringLiteral("HOME")
+#define kPathVarTemp                            QStringLiteral("TEMP")
+
+#define kGroupName                              "Esri/ArcGISRuntime/pathVariables"
+
+#define kPropertyPathVariablePrefix             kGroupName "."
+
+#define kSettingsGroupPathVariables             QStringLiteral(kGroupName)
+
+//------------------------------------------------------------------------------
+
+QString ZipHelper::resolvedPath(const QString& path, bool useFallbacks)
+{
+  if (path.startsWith(kPathVarPrefix) || path.startsWith(kPathHomePrefix))
+  {
+    QString resolvedPath = path;
+
+    QString varName;
+    int subLength = 1;
+
+    auto separatorIndex = resolvedPath.indexOf('/');
+    if (separatorIndex < 0)
+    {
+      separatorIndex = resolvedPath.indexOf('\\'); // Windows
+    }
+
+    if (separatorIndex < 0)
+    {
+      varName = resolvedPath.mid(1);
+      if (varName.isEmpty() && path.startsWith(kPathHomePrefix))
+      {
+        varName = kPathVarHome;
+      }
+
+      subLength = resolvedPath.length();
+    }
+    else if (separatorIndex < 2)
+    {
+      if (path.startsWith(kPathHomePrefix))
+      {
+        varName = kPathVarHome;
+      }
+
+      subLength = 1;
+    }
+    else
+    {
+      subLength = separatorIndex;
+      varName = resolvedPath.mid(1, subLength - 1);
+    }
+
+    QString varValue;
+
+    if (!varName.isEmpty())
+    {
+      varValue = pathVariable(varName, QString(), useFallbacks);
+
+      if (!pathVariableExists(varName, useFallbacks))
+      {
+#ifdef ZIP_LOGGING_ENABLED
+        qDebug() << "Nonexistent Path variable" << varName;
+#endif
+      }
+    }
+    else
+    {
+#ifdef ZIP_LOGGING_ENABLED
+      qDebug() << "Path variable name not specified";
+#endif
+    }
+
+    return resolvedPath.replace(0, subLength, varValue);
+  }
+
+  return path;
+}
+
+bool ZipHelper::setDateTime(const QString& filePath, const QDateTime& accessed, const QDateTime& modified)
```

Comment:
Is this called anywhere?

I have this comment for lots of ZipHelper. I don't know why we'd need to call `setDateTime` for unzipping files.

Reply:
ZipHelper::extractCurrentFile

This was apart of the original plugin for setting a timestamp for the file?

Reply:
I believe everything is made private that can be.